CMakeLists.txt

```diff
@@ -0,0 +1,191 @@
+    src/flac/encode.c
+    src/flac/foreign_metadata.c
+    src/flac/iffscan.c
+    src/flac/local_string_utils.c
+    src/flac/main.c
+    src/flac/utils.c
+    src/flac/vorbiscomment.c
+)
+
+set(FLAC_SRC_LIBFLAC
+    src/libFLAC/bitmath.c
+    src/libFLAC/bitreader.c
+    src/libFLAC/bitwriter.c
+    src/libFLAC/cpu.c
+    src/libFLAC/crc.c
+    src/libFLAC/fixed.c
+    src/libFLAC/float.c
+    src/libFLAC/format.c
+    src/libFLAC/lpc.c
+    src/libFLAC/md5.c
+    src/libFLAC/memory.c
+    src/libFLAC/metadata_iterators.c
+    src/libFLAC/metadata_object.c
+    src/libFLAC/ogg_decoder_aspect.c
+    src/libFLAC/ogg_encoder_aspect.c
+    src/libFLAC/ogg_helper.c
+    src/libFLAC/ogg_mapping.c
+    src/libFLAC/stream_decoder.c
+    src/libFLAC/stream_encoder.c
+    src/libFLAC/stream_encoder_framing.c
+    src/libFLAC/window.c
+)
+
+set(FLAC_SRC_SHARE
+)
+
+if (FLAC_USE_AVX)
+    list(APPEND FLAC_SRC_LIBFLAC
+        src/libflac/lpc_intrin_avx2.c
+        src/libflac/stream_encoder_intrin_avx2.c
+    )
+endif()
+
+if (FLAC_USE_SSE)
+    list(APPEND FLAC_SRC_LIBFLAC
+        src/libflac/lpc_intrin_sse.c
+    )
+endif()
+
+if (FLAC_USE_SSE2)
+    list(APPEND FLAC_SRC_LIBFLAC
+        src/libflac/lpc_intrin_sse2.c
+        src/libflac/stream_encoder_intrin_sse2.c
+    )
+endif()
+
+if (FLAC_USE_SSE41)
+    list(APPEND FLAC_SRC_LIBFLAC
+        src/libflac/lpc_intrin_sse41.c
+        src/libflac/stream_encoder_intrin_ssse3.c
+    )
+endif()
+
+
+add_definitions(-DFLAC__HAS_X86INTRIN)
+
+if (FLAC_USE_ASM)
+    add_definitions(-DFLAC__HAS_NASM)
+
+    if (FLAC_USE_3DNOW)
+        add_definitions(-DFLAC__USE_3DNOW)
+    endif()
+    
+    if (FLAC_CPU_X86_64)
+        add_definitions(-DFLAC__CPU_X86_64)
+    else()
+        add_definitions(-DFLAC__CPU_IA32)
+        if (FLAC_USE_SSE OR FLAC_USE_SSE2 OR FLAC_USE_SSE41)
+            add_definictions(-DFLAC__SSE_OS)
+        endif()
+
+        set(FLAC_SRC_LIBFLAC_NASM
+            src/libflac/ia32/cpu_asm.nasm
+            src/libflac/ia32/fixed_asm.nasm
+            src/libflac/ia32/lpc_asm.nasm
+        )
+    endif()
+else()
+    add_definitions(-DFLAC__NO_ASM)
+endif()
+
+add_definitions(-DFLAC__ALIGN_MALLOC_DATA)
+
+if (FLAC_USE_ASM)
+    nasm_command(FLAC_SRC_LIBFLAC_NASM FLAC_ASM_OUT)
+endif()
+
+if(WIN32)
+    list(APPEND FLAC_SRC_SHARE
+        src/share/win_utf8_io/win_utf8_io.c
+    )
+endif()
+
+set(FLAC_SOURCES ${FLAC_SRC_LIBFLAC} ${FLAC_SRC_SHARE})
+
+include_directories("${FLAC_SOURCE_DIR}/include")
+include_directories("${FLAC_SOURCE_DIR}/src/libflac/include")
+
+add_library(FLAC SHARED
+    ${FLAC_SOURCES}
+    ${FLAC_ASM_OUT}
+)
+
+if(WIN32)
+    set_target_properties(FLAC PROPERTIES DEFINE_SYMBOL "DLL_EXPORT")
+#   set_property(TARGET FLAC APPEND_STRING PROPERTY
+#       COMPILE_FLAGS "/FI\"${FLAC_SOURCE_DIR}/src/win32_dll.h\""
+#   )
+endif()
+
+target_link_libraries (FLAC
+    ${OGG_LIBRARIES}
+)
+
+set_target_properties(FLAC
+    PROPERTIES PREFIX "" OUTPUT_NAME "libFLAC"
```

Comment:
This seems odd, considering that had you not changed it, the default prefix would have been `lib` and the default output name would have been the name of the target, `FLAC`, so you'd still end up with `libFLAC.so`.

Is there any reason why you wanted to specify this (perhaps because of how things work on windows)?

Reply:
I don't really remember my reasons for these changes, if they were mine to begin with. I largely based this script on one composed by @soulik.